lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -1280,15 +1280,61 @@ void HdVP2Mesh::_InitRepr(const TfToken& reprToken, HdDirtyBits* dirtyBits)
 
         switch (desc.geomStyle) {
         case HdMeshGeomStyleHull:
-            break; // Creating the hull render items requires geom subsets from the topology, and we
-                   // can't access that here.
+            // Creating the smoothHull hull render items requires geom subsets from the topology,
+            // and we can't access that here.
+#ifdef HAS_DEFAULT_MATERIAL_SUPPORT_API
+            if (reprToken == HdVP2ReprTokens->defaultMaterial) {
+                // But default material mode does not use geom subsets, so we create the render item
+                renderItem = _CreateSmoothHullRenderItem(renderItemName);
+                renderItem->setDefaultMaterialHandling(
+                    MRenderItem::DrawOnlyWhenDefaultMaterialActive);
+                renderItem->setShader(_delegate->Get3dDefaultMaterialShader());
+            }
+#endif
+            break;
         case HdMeshGeomStyleHullEdgeOnly:
-            // The smoothHull repr uses the wireframe item for selection
-            // highlight only.
+            // The smoothHull repr uses the wireframe item for selection highlight only.
+#ifdef HAS_DEFAULT_MATERIAL_SUPPORT_API
+            if (reprToken == HdReprTokens->smoothHull
+                || reprToken == HdVP2ReprTokens->defaultMaterial) {
+                // Share selection highlight render item between smoothHull and defaultMaterial:
+                bool                        foundShared = false;
```

Comment:
If we end up using this pattern to share more render items then this seems like a good candidate to be refactored into a method.